### PR TITLE
[inference] add smoothquant llama attention

### PR DESCRIPTION
**Pull request by Xu-Kai**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [ ] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?
- add llama attention for smoothquant

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [ ] I have performed a self-review of my code
- [x] I have added thorough tests.
- [ ] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: colossalai/inference/quant/smoothquant/models/smoothquant_layer.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        self.pv_bmm = BMM_S8T_S8N_S8T(1.0)
+
+        self.k_proj = W8A8B8O8Linear(hidden_size, hidden_size)
+        self.v_proj = W8A8B8O8Linear(hidden_size, hidden_size)
+        self.q_proj = W8A8B8O8Linear(hidden_size, hidden_size)
+        self.out_proj = W8A8BFP32OFP32Linear(hidden_size, hidden_size)
+
+        self.q_output_scale = torch.tensor([1.0])
+        self.k_output_scale = torch.tensor([1.0])
+        self.rotary_output_scale = torch.tensor([1.0])
```
Comment (yuanheng-zhao) [inline colossalai/inference/quant/smoothquant/models/smoothquant_layer.py:43]: Just wonder: these scales are always the same as 1.0?

File: tests/test_smoothquant/test_llama_attention.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
+import pytest
+import torch
+from packaging import version
+
+from colossalai.inference.quant.smoothquant.models.smoothquant_layer import LLamaSmoothquantAttention
+from colossalai.kernel.triton import int8_rotary_embedding_fwd
+
+try:
+    from colossalai.inference.quant.smoothquant.models.smoothquant_layer import LLamaSmoothquantAttention
+    from colossalai.kernel.triton import int8_rotary_embedding_fwd
```
Comment (yuanheng-zhao) [inline tests/test_smoothquant/test_llama_attention.py:10]: Duplicated

File: tests/test_smoothquant/test_llama_attention.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+    HAS_TRITON = True
+except ImportError:
+    HAS_TRITON = False
+    print("please install triton from https://github.com/openai/triton")
+
+TRITON_CUDA_SUPPORT = version.parse(torch.version.cuda) > version.parse("11.4")
+
+TRITON_CUDA_SUPPORT = version.parse(torch.version.cuda) > version.parse("11.4")
```
Comment (yuanheng-zhao) [inline tests/test_smoothquant/test_llama_attention.py:19]: remove duplicated lines

File: colossalai/inference/quant/smoothquant/models/smoothquant_layer.py [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+    def pack(
```
Comment (yuanheng-zhao) [inline colossalai/inference/quant/smoothquant/models/smoothquant_layer.py:45]: Is the function required in future ? (seems not to be used in this PR) What 's the purpose of the function

File: colossalai/inference/quant/smoothquant/models/smoothquant_layer.py (hunk 1 above)
Comment (Xu-Kai) [reply colossalai/inference/quant/smoothquant/models/smoothquant_layer.py:43]: Just for init. Update values in the pack function.

File: colossalai/inference/quant/smoothquant/models/smoothquant_layer.py (hunk 4 above)
Comment (Xu-Kai) [reply colossalai/inference/quant/smoothquant/models/smoothquant_layer.py:45]: It used to get int8 weight and other values from original models.

File: tests/test_smoothquant/test_llama_attention.py (hunk 2 above)
Comment (Xu-Kai) [reply tests/test_smoothquant/test_llama_attention.py]: Removed.

File: tests/test_smoothquant/test_llama_attention.py (hunk 3 above)
Comment (Xu-Kai) [reply tests/test_smoothquant/test_llama_attention.py:22]: Removed.

**Review by yuanheng-zhao — APPROVED** [review]
LGTM

**Pull request merged**